src/Component/Auth/SignUp/index.js

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from "react";
+import { Link, Redirect} from "react-router-dom";
+import { signUpRequest } from "../../../Actions";
+import { connect } from "react-redux";
+
+import Header from "../../../Component/CommonComponent/Header";
+import Footer from "../../../Component/CommonComponent/Footer";
+
+import "./style.css";
+
+class SignUp extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      first_name: "",
+      last_name: "",
+      gender: "",
+      address: "",
+      email: "",
+      password: ""
+    };
+  }
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+  onSubmit = (e) => {
+    e.preventDefault();
+    const signUpData = {
+      first_name: this.state.first_name,
+      last_name: this.state.last_name,
+      gender: this.state.gender,
+      address: this.state.address,
+      email: this.state.email,
+      password: this.state.password
+    };
+    this.props.signUpRequest(signUpData, this.props.history);
+  };
+  render() {
+    if (this.props.register_user && this.props.register_user.success) {
```

Comment:
@thealitahir This logic should be wrapped on routes layer. Something like protected routes.